### PR TITLE
[USM] log.Warn() and disable goTLS if runtime compilation is not available

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

log.Warn() and disable goTLS if runtime compilation is not available
because it fallback to prebuild or not enabled

### Motivation

Avoid ugly log entry on customer where it failed for each golang process
`1683903668423427215 [Debug] could not hook new binary "....golang-process" for process 27727: error while attaching hooks: could not add return hook to function "crypto/tls.(*Conn).Read" in offset 4954500 due to: couldn't find program {UID: EBPFFuncName:uprobe__crypto_tls_Conn_Read__return EBPFSection:}: unknown section or eBPF function name
`
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/tracer.go [hunk 1]
```diff
@@ -843,6 +843,13 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer, bpfTelemetry *net
 	sockFDMap := tracer.GetMap(probes.SockByPidFDMap)
 	connectionProtocolMap := tracer.GetMap(probes.ConnectionProtocolMap)
 
+	if tracer.Type() != connection.TracerTypeKProbeRuntimeCompiled {
+		if c.EnableGoTLSSupport {
+			log.Warnf("disabling USM goTLS support as goTLS require kprobe runtime compilation")
```
Comment (hmahmood) [inline pkg/network/tracer/tracer.go:848]: require -> requires

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/tracer.go [hunk 2]
```diff
@@ -843,6 +843,13 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer, bpfTelemetry *net
 	sockFDMap := tracer.GetMap(probes.SockByPidFDMap)
 	connectionProtocolMap := tracer.GetMap(probes.ConnectionProtocolMap)
 
+	if tracer.Type() != connection.TracerTypeKProbeRuntimeCompiled {
```
Comment (guyarb) [inline pkg/network/tracer/tracer.go:846]: what about CO-RE?

Comment (nplanel) [reply pkg/network/tracer/tracer.go]: I added the check for CORE

File: pkg/network/tracer/tracer.go (hunk 1 above)
Comment (nplanel) [reply pkg/network/tracer/tracer.go]: done

File: pkg/network/tracer/tracer.go [hunk 3]
```diff
@@ -843,6 +843,13 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer, bpfTelemetry *net
 	sockFDMap := tracer.GetMap(probes.SockByPidFDMap)
 	connectionProtocolMap := tracer.GetMap(probes.ConnectionProtocolMap)
 
+	if tracer.Type() != connection.TracerTypeKProbeRuntimeCompiled && tracer.Type() != connection.TracerTypeKProbeCORE {
+		if c.EnableGoTLSSupport {
+			log.Warnf("disabling USM goTLS support as goTLS requires kprobe runtime compilation or CO-RE")
+			c.EnableGoTLSSupport = false
+		}
+	}
```
Comment (guyarb) [inline pkg/network/tracer/tracer.go:851]:
```suggestion
	if c.EnableGoTLSSupport && tracer.Type() != connection.TracerTypeKProbeRuntimeCompiled && tracer.Type() != connection.TracerTypeKProbeCORE {
 	 	log.Warnf("disabling USM goTLS support as goTLS requires kprobe runtime compilation or CO-RE")
	 	c.EnableGoTLSSupport = false
	}
```

File: pkg/network/tracer/tracer.go [hunk 4]
```diff
@@ -843,6 +843,13 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer, bpfTelemetry *net
 	sockFDMap := tracer.GetMap(probes.SockByPidFDMap)
 	connectionProtocolMap := tracer.GetMap(probes.ConnectionProtocolMap)
 
+	if tracer.Type() != connection.TracerTypeKProbeRuntimeCompiled && tracer.Type() != connection.TracerTypeKProbeCORE {
+		if c.EnableGoTLSSupport {
+			log.Warnf("disabling USM goTLS support as goTLS requires kprobe runtime compilation or CO-RE")
```
Comment (guyarb) [inline pkg/network/tracer/tracer.go:848]:
```suggestion
			log.Warn("disabling USM goTLS support as goTLS requires runtime compilation or CO-RE")
```

**Pull request merged**